server/routes/api/import_detection/detection_types.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Classes (Enum, dataclass) used in the the various import_detection modules."""
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Sequence, Union
+
+
+# Helper Enums and Data Classes for Detection.
+class MappingType(Enum):
+    """Supported types of column structure."""
+    COLUMN = "column"
+    COLUMN_HEADER = "columnHeader"
+
+
+class MappedThing(Enum):
+    """Supported type of Thing being mapped."""
+    DATE = "Date"
+    PLACE = "Place"
```

Comment:
Is it worth using Enums https://docs.python.org/3/library/enum.html ?

For example, `mapping: Dict[str, MappingVal] = {}` can perhaps use `MappedThing` in place of `str`

Reply:
Not sure I follow. The reason mapping is using a str key is that the json.dumps() only really works on these structs if it is a primitive type or a dict like structure. Enums don't appear to work :(

Reply:
Ah, got it.